qiskit_experiments/calibration_management/basis_gate_library.py

```diff
@@ -289,3 +321,117 @@ def default_values(self) -> List[DefaultCalValue]:
                     defaults.append(DefaultCalValue(value, param.name, tuple(), name))
 
         return defaults
+
+
+class EchoedCrossResonance(BasisGateLibrary):
+    """A fixed-frequency transmon gate library with echoed cross-resonance gates.
+
+    This library extends the FixedFrequencyTransmon library by adding support for
+    cross-resonance gates.
+    """
+
+    __default_values__ = {
+        "cr_duration": 640,
+        "cr_width": 384,
+        "cr_σ": 64,
+        "cr_amp": 0.5,
+        "rot_amp": 0.0,
+    }
+
+    def __init__(
+        self,
+        single_qubit_library: BasisGateLibrary,
+        pulse_on_target: bool = True,
+        basis_gates: Optional[List[str]] = None,
+        default_values: Optional[Dict] = None,
+    ):
+        """Setup the schedules.
+
+        Args:
+            single_qubit_library: The library with the single-qubit gates from which
+                the X-gate schedule will be called to build the echo. This library needs
+                to supply a single-qubit schedule named ``"x"``.
+            pulse_on_target: A boolean to indicate if a rotary/cancellation tones are
+                used in the CR gate. If this value is set to False then no pulse is played
+                on the target qubit during the cross-resonance pulse.
+            basis_gates: The basis gates to generate.
+            default_values: Default values for the parameters this dictionary can contain
+                the following keys: "duration", "amp", "β", and "σ". If "σ" is not provided
+                this library will take one fourth of the pulse duration as default value.
+        """
+        self._rotary = pulse_on_target
+        super().__init__(basis_gates, default_values, [single_qubit_library])
+
+    @property
+    def __supported_gates__(self) -> Dict[str, int]:
+        """The gates that this library supports."""
+        return {"ecr": 2, "cr90p": 2, "cr90m": 2}
+
+    def _build_schedules(
+        self,
+        basis_gates: Set[str],
+        other_libraries: Optional[List["BasisGateLibrary"]] = None,
+    ) -> Dict[str, ScheduleBlock]:
+        """Build the schedules of the library."""
+        if "x" not in other_libraries[0]:
+            raise CalibrationError("x gate is required to build cross-resonance schedules.")
+
+        target = pulse.DriveChannel(Parameter("ch1"))
+        cr_chan = pulse.ControlChannel(Parameter("ch0.1"))
+
+        cr_amp = Parameter("cr_amp")
+        rot_amp = Parameter("rot_amp")
+        cr_sigma = Parameter("cr_σ")
+        cr_duration = Parameter("cr_duration")
+        cr_width = Parameter("cr_width")
+
+        cr90p = pulse.GaussianSquare(
+            duration=cr_duration, amp=cr_amp, sigma=cr_sigma, width=cr_width, name="cr90p"
+        )
+
+        cr90m = pulse.GaussianSquare(
+            duration=cr_duration, amp=-cr_amp, sigma=cr_sigma, width=cr_width, name="cr90m"
+        )
+
+        rot90p = pulse.GaussianSquare(
```

Comment:
Similar issue about the name...the target qubit is not rotated by 90. Maybe `cr45p_target` or just `target_p` (since this name is not getting referenced in a larger name space like the schedule names are) or something like that?

Reply:
Done here: [726cfe0](https://github.com/Qiskit/qiskit-experiments/pull/725/commits/726cfe0b28f48e60545a7a2a7eefd66d14eee7a2)